### PR TITLE
INT-3902: XML: SmartLifecycle, Order Attributes

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INT-3902

### CODE REVIEW

File: spring-integration-xml/src/test/java/org/springframework/integration/xml/config/XPathRouterTests-context.xml [hunk 1]
```diff
@@ -7,6 +7,15 @@
 		http://www.springframework.org/schema/integration http://www.springframework.org/schema/integration/spring-integration.xsd
 		http://www.springframework.org/schema/integration/xml http://www.springframework.org/schema/integration/xml/spring-integration-xml.xsd">
 
+	<int-xml:xpath-router id="parseOnly" input-channel="xpathRouterEmptyChannel"
+			order="2"
+			auto-startup="false"
+			role="foo"
+			timeout="123"
```
Comment (artembilan) [inline spring-integration-xml/src/test/java/org/springframework/integration/xml/config/XPathRouterTests-context.xml]: I wonder if we should deprecated this and introduce a new `send-timeout` one.
To be consistent with all other `messagingTemplate-aware` components, like `splitter`, `filter` etc..

Comment (garyrussell) [inline spring-integration-xml/src/test/java/org/springframework/integration/xml/config/XPathRouterTests-context.xml]: :+1:

**Comment by garyrussell** [comment]
Pushed

**Comment by artembilan** [comment]
Yep! LGTM - Merging...

**Comment by artembilan** [comment]
Here is a polishing for your confirmation: https://github.com/artembilan/spring-integration/commit/c644950673fc8d125fd80395934788b2bee16202.
I decided to introduce `setSendTimeout` as well.

**Comment by garyrussell** [comment]
I thought about doing that but I was lazy :smile: 

LGTM

**Comment by artembilan** [comment]
Yeah... That really was some waste of time...
Like all that XSD work :smile: 

Merging anyway.
Thank you for confirmation!

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/d684dc4d0e17a3dd0c840e9f0c792fd14dec9b4b.

**Pull request closed without merging**